### PR TITLE
sgd refactor

### PR DESCRIPTION
**Pull request by danqi437**
- 拆分SGDClassifier为SGDClassifier/SGDRegressor
- 新增decay_epoch，decay_rate,strategy等参数

### CODE REVIEW

File: sml/linear_model/sgd.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        if self._decay_rate is not None:
+            rate = self._decay_rate ** math.floor(epoch_idx / self._decay_epoch)
+            return self._learning_rate * rate
+        else:
+            return self._learning_rate
+
+    def _fit(self, X: jax.Array, y: jax.Array, activation: Callable):
+        n_samples, n_features = X.shape
+        batch_size = min(self._batch_size, n_samples)
+        total_batch = int(n_samples / batch_size)
```
Comment (deadlywing) [inline sml/linear_model/sgd.py:166]: maybe `total_batch = int((n_samples + batch_size - 1) / batch_size)` is better?

File: sml/linear_model/sgd.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                strategy=self._strategy,
+            )
+
+            if (
+                self._epsilon > 0
+                and idx > 0
+                and _convergence(old_w, weights, self._eps_norm, self._eps_scale)
+            ):
+                # early stop
+                break
```
Comment (deadlywing) [inline sml/linear_model/sgd.py:190]: The early stop under SPU should be implemented with the JAX api.
You can refer to https://github.com/secretflow/sml/blob/9e84246941106e362f86c124d053d9ef129d725f/sml/linear_model/logistic.py#L261-L293

for an example.

File: sml/linear_model/sgd.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        self,
+        epochs: int,
+        learning_rate: float = 0.1,
+        batch_size: int = 1024,
+        penalty: str | Penalty = Penalty.NONE,
+        l2_norm: float = 0.5,
+        epsilon: float = 1e-3,
+        decay_epoch: int | None = None,
+        decay_rate: float | None = None,
+        sig_type: SigType = SigType.T1,
```
Comment (deadlywing) [inline sml/linear_model/sgd.py:220]: The  `sigmoid` approximation implemented in SML is slightly different from SecretFlow, see https://github.com/secretflow/secretflow/blob/main/secretflow/utils/sigmoid.py.

I suggest we align with the implementation in SecretFlow. For Logistic Regression (LR), it's essential to clamp/clip the sigmoid output to the [0, 1] range.

File: tests/linear_model/sgd_test.py
Comment (deadlywing) [inline tests/linear_model/sgd_test.py:1]: After successfully implementing early stopping, it is recommended to test whether early stopping is effective.

**Review by deadlywing — APPROVED** [review]
LGTM
